### ISSUE TITLE
Go inside a foloder using C-j, after bookmark-jump

### ISSUE THREAD

**Issue by DrWaleedAYousef** [emacs-helm/helm#2319]
**Is your feature request related to a problem? Please describe.**
When pressing C-x r b (to call `bookmark-jump`) it populates the bookmarks. Then, pressing `C-j` does not go inside a highlighted folder as in the mini-buffer of helm-find

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by manuel-uberti** [comment]
A couple of things:

- What is `C-j` bound to in your Emacs configuration? You can check by typing <kbd>C-h k C-j</kbd>.

- Does pressing <kbd>RET</kbd> open the bookmark?

**Comment by DrWaleedAYousef** [comment]
Thanks so much for fast reply. Inside the mini-buffe of the bookmark, `C-j` is bound to `(helm-execute-persistent-action &optional ATTR SPLIT)'. And yes, presssing enter opens the bookmark

**Comment by manuel-uberti** [comment]
If you do <kbd>M-x helm-filtered-bookmarks RET</kbd>, select your bookmark, and press <kbd>C-j</kbd>, do you get the result you are after?

If so, I suggest you use `helm-filtered-bookmarks` instead of regular `bookmark-jump` (that's what I use, by the way).

**Comment by DrWaleedAYousef** [comment]
Yes, it works partially. I mean it opens the bookmark in another buffer. However, I cannot navigate in the mini-buffer using the left/right arrow as in the mini-buffer of helm-find

**Comment by manuel-uberti** [comment]
Sorry, I don't understand what you are trying to do. 

When I use `helm-filtered-bookmarks` I see the list of my bookmarks, and I pick the one I am looking for by typing and letting Helm narrowing down to it. Then, I hit <kbd>RET</kbd> (<kbd>C-j</kbd> in your case).

Where does the navigation should be done here?

**Comment by DrWaleedAYousef** [comment]
Instead of hitting `C-j`, trying to navigate using the left/right arrow does not work as in the mini-buffer of helm-find.

**Comment by manuel-uberti** [comment]
Check what <kbd>\<left\></kbd> and <kbd>\<right\></kbd> are bound to. In `helm-filtered-bookmarks`, I have:

- <kbd>\<left\></kbd> bound to `helm-previous-source`
- <kbd>\<right\></kbd> bound to `helm-next-source`

I don't really see the point of using these keys in `helm-filtered-bookmarks` (TBH, I never use them at all when in Helm). More importantly, I don't think they serve the purpose you are after in `helm-filtered-bookmarks`.

It seems to me like you want a different behaviour for these keys here. If so, wait for @thierryvolpiatto to discuss this, because this as far as my knowledge goes. :)

**Comment by thierryvolpiatto** [comment]
Manuel Uberti <notifications@github.com> writes:

> Check what <left> and <right> are bound to. In helm-filtered-bookmarks, I have:
>
>   • <left> bound to helm-previous-source
>   • <right> bound to helm-next-source
>
> I don't really see the point of using these keys in
> helm-filtered-bookmarks (TBH, I never use them at all when in
> Helm). More importantly, I don't think they serve the purpose you are
> after in helm-filtered-bookmarks.
>
> It seems to me like you want a different behaviour for these keys
> here. If so, wait for @thierryvolpiatto to discuss this, because this
> as far as my knowledge goes. :)

I think what is asked is how to bookmark directories and then jump to them
in `helm-find-files`. Is it right?
If so this comes out of the box, go in `helm-find-files`, navigate to a
directory and hit `C-x r m`, then you can retrieve this bookmark either
in `helm-filtered-bookmarks`, or directly from helm-find-files by hitting
`C-x r b`, when you press RET you are in `helm-find-files` at your
directory.
If you have bookmarked directories from dired, this will not work (it is
not the same bookmark handler), when pressing RET you will be dropped in
dired, not in HFF.

-- 
Thierry

Get my Gnupg key:
gpg --keyserver pgp.mit.edu --recv-keys 59F29997

**Comment by manuel-uberti** [comment]
I was about to write that one can bookmark `helm-find-files` session too (which is something I do and find very handy). 

So, @DrWaleedAYousef, I think this way you can retain the usage you're looking for.

**Comment by DrWaleedAYousef** [comment]
> Manuel Uberti <notifications@github.com> writes:
> Check what <left> and <right> are bound to. In helm-filtered-bookmarks, I have: • <left> bound to helm-previous-source • <right> bound to helm-next-source I don't really see the point of using these keys in helm-filtered-bookmarks (TBH, I never use them at all when in Helm). More importantly, I don't think they serve the purpose you are after in helm-filtered-bookmarks. It seems to me like you want a different behaviour for these keys here. If so, wait for @thierryvolpiatto to discuss this, because this as far as my knowledge goes. :)
> I think what is asked is how to bookmark directories and then jump to them in `helm-find-files`. Is it right? If so this comes out of the box, go in `helm-find-files`, navigate to a directory and hit `C-x r m`, then you can retrieve this bookmark either in `helm-filtered-bookmarks`, or directly from helm-find-files by hitting `C-x r b`, when you press RET you are in `helm-find-files` at your directory. 

This is almost what I need; however, can I make `C-j` having the same effect as RET in this context. Also, as I will start from now on using `helm-filtered-bookmarks` instead, do I need now to rename the file `bookmarks` to `helm-bookmarks`. 

If you have bookmarked directories from dired, this will not work (it is not the same bookmark handler), when pressing RET you will be dropped in dired, not in HFF.
> […](#)
> -- Thierry Get my Gnupg key: gpg --keyserver pgp.mit.edu --recv-keys 59F29997

**Comment by manuel-uberti** [comment]
You don't need to rename `bookmarks` for `helm-filtered-bookmarks` to work.

As for <kbd>C-j</kbd>: assuming <kbd>RET</kbd> is bound to `helm-maybe-exit-minibuffer`, you'd have to bind <kbd>C-j</kbd> to that too (or instead, depending on what you'd want to do with <kbd>RET</kbd> then).

If I am not mistaken, this way you'll change it for other Helm commands too, because `helm-filtered-bookmarks` doesn't have a specific keymap (like, say, `helm-find-files-map`) you can leverage for this particular change.
